crypto/src/engine/blst/mod.rs

```diff
@@ -0,0 +1,132 @@
+mod g1;
+mod g2;
+mod scalar;
+
+use rayon::prelude::{
+    IndexedParallelIterator, IntoParallelIterator, IntoParallelRefIterator,
+    IntoParallelRefMutIterator, ParallelIterator,
+};
+use std::iter;
+
+use crate::{CeremonyError, Engine, ParseError, G1, G2};
+
+use self::{
+    g1::{batch_g1_projective_to_affine, G1BlstAffine, G1BlstProjective},
+    g2::{batch_g2_projective_to_affine, G2BlstAffine, G2BlstProjective},
+    scalar::{random_scalar, scalar_mul, ScalarBlst},
+};
+
+pub struct BLST;
+
+pub trait BLSTAlgebra {
+    fn mul(&self, scalar: &ScalarBlst) -> Self;
+    fn is_in_subgroup(&self) -> bool;
+}
+
+impl Engine for BLST {
+    fn add_entropy_g1(
+        entropy: [u8; 32],
+        powers: &mut [crate::G1],
+    ) -> Result<(), crate::CeremonyError> {
+        let tau = ScalarBlst::try_from(random_scalar(entropy))?;
```

Comment:
One option for BLST specifically is to use the `blst::blst_keygen_v5` [function from the BLST crate (docs)](https://docs.rs/blst/0.3.10/blst/fn.blst_keygen_v5.html) which comes from the BLS signature IETF specs as a way of getting randomness with sufficiently low bias.

TLDR is HKDF extract and expand out to 48 Bytes then modulo reduce.